### PR TITLE
Tweak "wrong # of generics" suggestions

### PR DESCRIPTION
**Pull request by estebank**
Fix incorrect suggestion, make verbose and change message to make more sense when it isn't a span label.

### CODE REVIEW

**Comment by oli-obk** [comment]
idk about the verbose version. The short version seems clearer to me as all the information is in one location, and the remove span doesn't even overlap with the error span

File: compiler/rustc_hir_analysis/src/errors/wrong_number_of_generic_args.rs [hunk 1]
```diff
@@ -888,7 +888,7 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         let comma = if args.len() > 0 { ", " } else { "" };
         let trait_path = self.tcx.def_path_str(trait_def_id);
         let method_name = self.tcx.item_name(self.def_id);
-        err.span_suggestion(
+        err.span_suggestion_verbose(
```
Comment (oli-obk) [inline compiler/rustc_hir_analysis/src/errors/wrong_number_of_generic_args.rs:891]: Did you forget this one?

Comment (estebank) [reply compiler/rustc_hir_analysis/src/errors/wrong_number_of_generic_args.rs:891]: Those correspond to these, https://github.com/rust-lang/rust/pull/127374/files/921de9d8eae16947c375febe0ab8709797b37119#diff-1d620ac138e59caa4ee1265c976245aea8bb80648f969e59af382d38bfd84dd9R14, which I could not find an example where they would be rendered on their own, but given how *long* the suggested code and span are, I'd prefer to leave these as verbose. (I personally find "help: frobnicate the expression: `very::long::code::<that, goes, const { on }>(and, on)`" to be quite hard to read.)

**Pull request merged**